configure.py

```diff
@@ -1420,6 +1424,106 @@ def bakefile_test_sources(file, sources):
 
     f.close()
 
+def gen_cmake(build_config, using_mods, cc, options):
+    def create_basic_rules(sources):
+        module = {'sources': {}, 'frameworks': set(), 'libs': set()}
+        for source in sources:
+            module['sources'][source] = {'isa_flags': set()}
+        return module
+
+    def update_rules(rules, using_mod):
+        needed = False
+        for source_path in rules['sources']:
+            for mod_source in using_mod.source:
+                if source_path == mod_source :
+                    needed = True
+                    for isa in using_mod.need_isa:
+                        rules['sources'][source_path]['isa_flags'].add(cc.isa_flags[isa])
+        if needed == True:
+            if options.os in using_mod.libs:
+                for lib in using_mod.libs[options.os]:
+                    rules['libs'].add(lib)
+            if options.os in using_mod.frameworks:
+                for framework in using_mod.frameworks[options.os]:
+                    rules['frameworks'].add(framework)
+
+    def generate_sources_list(fd, name, rules):
+        fd.write('set(' + name + '\n')
+        for source in rules['sources']:
+            fd.write('    ${CMAKE_CURRENT_LIST_DIR}/' + source[2:] + '\n')
```

Comment:
Cutting substrings from paths looks dangerous to me. I think [os.path.normpath](https://docs.python.org/2/library/os.path.html#os.path.normpath) is your friend

```
$ python -c "import os; print(os.path.normpath('.'))"
.
$ python -c "import os; print(os.path.normpath('ab.cpp'))"
ab.cpp
$ python -c "import os; print(os.path.normpath('./sub/ab.cpp'))"
sub/ab.cpp

However, this will result in backward slashes on Windows and I don't know if this causes problems with cmake